apps/prairielearn/src/pages/elementFiles/elementFiles.js

```diff
@@ -24,67 +23,81 @@ const sql = sqldb.loadSqlEquiv(import.meta.url);
-      throw new HttpStatusError(404, 'Unable to serve that file');
-    }
+export default function (options = { publicQuestionEndpoint: false, coreElements: false }) {
+  const router = Router({ mergeParams: true });
+  router.get(
+    '/*',
+    asyncHandler(async (req, res) => {
+      const filename = req.params[0];
+      const pathSpl = path.normalize(filename).split('/');
+      const valid =
+        pathSpl[1] === CLIENT_FOLDER ||
+        EXTENSION_WHITELIST.some((extension) => filename.endsWith(extension));
+      if (!valid) {
+        throw new HttpStatusError(404, 'Unable to serve that file');
+      }
 
-    // If the route includes a `cachebuster` param, we'll set the `immutable`
-    // and `maxAge` options on the `Cache-Control` header. This router is
-    // mounted twice - one with the cachebuster in the URL, and once without it
-    // for backwards compatibility. See `server.js` for more details.
-    //
-    // As with `/assets/`, we assume that element files are likely to change
-    // when running in dev mode, so we skip caching entirely in that case.
-    const isCached = !!req.params.cachebuster && !config.devMode;
-    const sendFileOptions = {
-      immutable: isCached,
-      maxAge: isCached ? '31536000s' : 0,
-    };
+      // If the route includes a `cachebuster` param, we'll set the `immutable`
+      // and `maxAge` options on the `Cache-Control` header. This router is
+      // mounted twice - one with the cachebuster in the URL, and once without it
+      // for backwards compatibility. See `server.js` for more details.
+      //
+      // As with `/assets/`, we assume that element files are likely to change
+      // when running in dev mode, so we skip caching entirely in that case.
+      const isCached = !!req.params.cachebuster && !config.devMode;
+      const sendFileOptions = {
+        immutable: isCached,
+        maxAge: isCached ? '31536000s' : 0,
+      };
 
-    if (isCached) {
-      // `middlewares/cors.js` disables caching for all routes by default.
-      // We need to remove this header so that `res.sendFile` can set it
-      // correctly.
-      res.removeHeader('Cache-Control');
-    }
+      if (isCached) {
+        // `middlewares/cors.js` disables caching for all routes by default.
+        // We need to remove this header so that `res.sendFile` can set it
+        // correctly.
+        res.removeHeader('Cache-Control');
+      }
 
-    let elementFilesDir;
-    if (res.locals.course) {
-      // Files should be served from the course directory
-      let question_course;
-      if (req.params.producing_course_id) {
-        const producing_course_id = z.string().parse(req.params.producing_course_id);
-        const has_shared_question = await sqldb.queryRow(
-          sql.select_has_shared_question,
-          { consuming_course_id: res.locals.course.id, producing_course_id },
+      let elementFilesDir;
+      if (options.publicQuestionEndpoint && !options.coreElements) {
```

Comment:
Very much nitpicking here, but I wonder if ordering the conditionals like this would be easier to understand:

```ts
if (options.coreElements) {
  // ...
} else if (options.publicQuestionEndpoint) {
  // ...
} else {
  // ...
}
```

Feel free to reject this, just wondering out loud.

Reply:
ya, I think you're right, that's easier to understand